be/src/exec/vectorized/jdbc_scanner.cpp

```diff
@@ -390,10 +391,27 @@ Status JDBCScanner::_fill_chunk(jobject jchunk, size_t num_rows, ChunkPtr* chunk
         for (size_t i = 0; i < _slot_descs.size(); i++) {
             jobject jcolumn = helper.list_get(jchunk, i);
             LOCAL_REF_GUARD_ENV(env, jcolumn);
-            auto result_column = _result_chunk->columns()[i].get();
-            auto st = helper.get_result_from_boxed_array(_result_column_types[i], result_column, jcolumn, num_rows);
+            auto& result_column = _result_chunk->columns()[i];
+            auto st =
+                    helper.get_result_from_boxed_array(_result_column_types[i], result_column.get(), jcolumn, num_rows);
             RETURN_IF_ERROR(st);
-            down_cast<NullableColumn*>(result_column)->update_has_null();
+            // check data's length for varchar type
+            if (_slot_descs[i]->type().type == TYPE_VARCHAR) {
```

Comment:
we already have a similar function StringColumnWriter::check_string_lengths. we should try to use the same function

Reply:
> we already have a similar function StringColumnWriter::check_string_lengths. we should try to use the same function

This function itself is not exposed to the outside. although these two functions are similar, but the required params and the returned error message content are different. There is no need to reuse it for reuse, right?